### PR TITLE
Dynamic extraction of Manifest Data

### PR DESCRIPTION
**Pull request by black-eagle17**
Greetings,
     I am new to open source development and programming overall. I managed to get the manifest data printing using the derivation technique suggested. However, I am having a little trouble figuring out how I would iterate through the list of supported platforms to get all of them to print. I hope you can review my code and give me pointers/feedback so I can learn from my mistakes.

### CODE REVIEW

**Comment by rdagumampan** [comment]
Hi @black-eagle17 , thank you very much for you contribution. Every contribution counts, so we highly appreciate it. For your question, you can try `Reflection` if you can extract all implementations of `IDataService` and create dynamic instance of the classes. You can also use `SUPPORTED_DATABASES` enums as keys to find the implementations faster. HT/Rodel

**Comment by black-eagle17** [comment]
Thank you for the feedback, I read up on reflection and it is now implemented in the code. I think I am ready for a PR review.

**Comment by rdagumampan** [comment]
Thank you very much @black-eagle17. I am reviewing your PR now but may take a while. There were many changes that doesnt seem to be reall change, maybe just some encoding differences. I will try to do Beyond Compare of the two branches and see the actual text change.

**Comment by black-eagle17** [comment]
@rdagumampan Thank you, any suggestions on how I could ensure a consistent encoding. I'm on Linux so it might be how I checkout my line endings.

**Comment by rdagumampan** [comment]
Hi @black-eagle17 , i think its the encoding. When using Beyond Compare, all of your changes looks ok. I think I can pull this change over after I hear back from you on the other comments.

image

File: yuniql-cli/CommandLineService.cs [hunk 1]
```diff
@@ -281,110 +281,85 @@ public int RunDestroyOption(DestroyOption opts)
...
+                 
+                //iterate through supported db and get manifest data
+                var fields = typeof(SUPPORTED_DATABASES).GetFields();
+                foreach (var field in fields)
+                {
+                    if(field.Name.ToLower() != SUPPORTED_DATABASES.MARIADB) //extra check since mariadb is not yet fully supported[SHOULD CHANGE IN THE FUTURE]
+                    {
+                        var _dataService = _dataServiceFactory.Create(field.Name);
+                        var _manifestData =_dataService.GetManifestData();
+                        _manifestData.printData();
```
Comment (rdagumampan) [inline yuniql-cli/CommandLineService.cs:315]: Lets move the logic of this into the `CommandLineService` so this class remains just plain object without any behavior. How we format into the output stream would be responsibility of the CLI classes.

**Review by rdagumampan — CHANGES_REQUESTED** [review]
See me comments below.

**Pull request merged**